### PR TITLE
[HOPSWORKS-1418] Pytorch 1 3

### PR DESCRIPTION
**Pull request by jimdowling**
https://logicalclocks.atlassian.net/browse/HOPSWORKS-1418
Bump pytorch, numpy versions.

### CODE REVIEW

File: attributes/default.rb [hunk 1]
```diff
@@ -11,11 +11,11 @@
 default['tensorflow']['rocm']['version']         = "1.14.0"
 
 default["tensorflow"]['serving']["version"]      = "1.14.0"
-default["cudatoolkit"]["version"]                = "10.0"
-default["pytorch"]["version"]                    = "1.0.1"
-default["pytorch"]["python2"]["build"]           = "py2.7_cuda10.0.130_cudnn7.4.2_2"
-default["pytorch"]["python3"]["build"]           = "py3.6_cuda10.0.130_cudnn7.4.2_2"
-default["torchvision"]["version"]                = "0.2.1"
+default["cudatoolkit"]["version"]                = "10.1"
```
Comment (robzor92) [inline attributes/default.rb]: Put back to 10.0 as we use a shared CUDA installation on the GPU machines which is 10.0.

File: recipes/default.rb [hunk 2]
```diff
@@ -330,16 +330,12 @@
...
-      if [ "#{python}" == "2.7" ] ; then
-        ${CONDA_DIR}/bin/conda install -y -n ${ENV} -c ${PYTORCH_CHANNEL} pytorch=#{node['pytorch']['version']}=#{node["pytorch"]["python2"]["build"]} torchvision=#{node['torchvision']['version']} cudatoolkit=#{node['cudatoolkit']['version']}
-      else
-        ${CONDA_DIR}/bin/conda install -y -n ${ENV} -c ${PYTORCH_CHANNEL} pytorch=#{node['pytorch']['version']}=#{node["pytorch"]["python3"]["build"]} torchvision=#{node['torchvision']['version']} cudatoolkit=#{node['cudatoolkit']['version']}
-      fi
-      ${CONDA_DIR}/bin/conda remove -y -n ${ENV} cudatoolkit=#{node['cudatoolkit']['version']} --force
+    if [ "#{python}" == "2.7" ] ; then
+      ${CONDA_DIR}/bin/conda install -y -n ${ENV} -c ${PYTORCH_CHANNEL} pytorch=#{node['pytorch']['version']}=#{node["pytorch"]["python2"]["build"]} torchvision=#{node['torchvision']['version']} cudatoolkit=#{node['cudatoolkit']['version']}
     else
-      ${CONDA_DIR}/bin/conda install -y -n ${ENV} -c ${PYTORCH_CHANNEL} pytorch-cpu=#{node['pytorch']['version']} torchvision-cpu=#{node['torchvision']['version']}
```
Comment (robzor92) [inline recipes/default.rb]: Put back the -cpu annotations suffix. We do this to save space.

File: recipes/default.rb [hunk 3]
```diff
@@ -330,16 +330,12 @@
...
+      ${CONDA_DIR}/bin/conda install -y -n ${ENV} -c ${PYTORCH_CHANNEL} pytorch=#{node['pytorch']['version']}=#{node["pytorch"]["python3"]["build"]} torchvision=#{node['torchvision']['version']} cudatoolkit=#{node['cudatoolkit']['version']}
```
Comment (robzor92) [inline recipes/default.rb]: Put back the -cpu suffix, we do this to save space.

File: attributes/default.rb (hunk 1 above)
Comment (jimdowling) [reply attributes/default.rb]: ok.

File: recipes/default.rb (hunk 3 above)
Comment (jimdowling) [reply recipes/default.rb]: There is no more -cpu fix version of pytorch. In 1.2 (i think), they now have just one version of pytorch (-cpu and -tpu packages are gone)

File: recipes/default.rb [hunk 4]
```diff
@@ -332,18 +332,21 @@
 
     if [ $TENSORFLOW_LIBRARY_SUFFIX == "-gpu" ] ; then
       if [ "#{python}" == "2.7" ] ; then
-        ${CONDA_DIR}/bin/conda install -y -n ${ENV} -c ${PYTORCH_CHANNEL} pytorch=#{node['pytorch']['version']}=#{node["pytorch"]["python2"]["build"]} torchvision=#{node['torchvision']['version']} cudatoolkit=#{node['cudatoolkit']['version']}
+        ${CONDA_DIR}/bin/conda install -y -n ${ENV} -c ${PYTORCH_CHANNEL} pytorch=#{node['pytorch']['version']}=#{node["pytorch"]["python2"]["build"]} torchvision=#{node['torchvision']['version']} 
+#cudatoolkit=#{node['cudatoolkit']['version']}
       else
-        ${CONDA_DIR}/bin/conda install -y -n ${ENV} -c ${PYTORCH_CHANNEL} pytorch=#{node['pytorch']['version']}=#{node["pytorch"]["python3"]["build"]} torchvision=#{node['torchvision']['version']} cudatoolkit=#{node['cudatoolkit']['version']}
+        ${CONDA_DIR}/bin/conda install -y -n ${ENV} -c ${PYTORCH_CHANNEL} pytorch=#{node['pytorch']['version']}=#{node["pytorch"]["python3"]["build"]} torchvision=#{node['torchvision']['version']}
```
Comment (robzor92) [inline recipes/default.rb]: @jimdowling did you push the code where you don't install cudatoolkit?

Comment (robzor92) [reply recipes/default.rb]: Ah I see it is commented out, can you remove it in that case?

File: recipes/default.rb [hunk 5]
```diff
@@ -332,18 +332,21 @@
...
-        ${CONDA_DIR}/bin/conda install -y -n ${ENV} -c ${PYTORCH_CHANNEL} pytorch=#{node['pytorch']['version']}=#{node["pytorch"]["python2"]["build"]} torchvision=#{node['torchvision']['version']} cudatoolkit=#{node['cudatoolkit']['version']}
+        ${CONDA_DIR}/bin/conda install -y -n ${ENV} -c ${PYTORCH_CHANNEL} pytorch=#{node['pytorch']['version']}=#{node["pytorch"]["python2"]["build"]} torchvision=#{node['torchvision']['version']} 
+#cudatoolkit=#{node['cudatoolkit']['version']}
       else
-        ${CONDA_DIR}/bin/conda install -y -n ${ENV} -c ${PYTORCH_CHANNEL} pytorch=#{node['pytorch']['version']}=#{node["pytorch"]["python3"]["build"]} torchvision=#{node['torchvision']['version']} cudatoolkit=#{node['cudatoolkit']['version']}
+        ${CONDA_DIR}/bin/conda install -y -n ${ENV} -c ${PYTORCH_CHANNEL} pytorch=#{node['pytorch']['version']}=#{node["pytorch"]["python3"]["build"]} torchvision=#{node['torchvision']['version']} 
+#cudatoolkit=#{node['cudatoolkit']['version']}
       fi
-      ${CONDA_DIR}/bin/conda remove -y -n ${ENV} cudatoolkit=#{node['cudatoolkit']['version']} --force
+#      ${CONDA_DIR}/bin/conda remove -y -n ${ENV} cudatoolkit --force
```
Comment (robzor92) [inline recipes/default.rb]: also remove this line

File: recipes/default.rb [hunk 6]
```diff
@@ -332,18 +332,18 @@
...
+        ${CONDA_DIR}/bin/conda install -y -n ${ENV} -c ${PYTORCH_CHANNEL} pytorch=#{node['pytorch']['version']}=#{node["pytorch"]["python2"]["build"]} torchvision=#{node['torchvision']['version']} 
       else
-        ${CONDA_DIR}/bin/conda install -y -n ${ENV} -c ${PYTORCH_CHANNEL} pytorch=#{node['pytorch']['version']}=#{node["pytorch"]["python3"]["build"]} torchvision=#{node['torchvision']['version']} cudatoolkit=#{node['cudatoolkit']['version']}
+        ${CONDA_DIR}/bin/conda install -y -n ${ENV} -c ${PYTORCH_CHANNEL} pytorch=#{node['pytorch']['version']}=#{node["pytorch"]["python3"]["build"]} torchvision=#{node['torchvision']['version']} 
       fi
-      ${CONDA_DIR}/bin/conda remove -y -n ${ENV} cudatoolkit=#{node['cudatoolkit']['version']} --force
     else
-      ${CONDA_DIR}/bin/conda install -y -n ${ENV} -c ${PYTORCH_CHANNEL} pytorch-cpu=#{node['pytorch']['version']} torchvision-cpu=#{node['torchvision']['version']}
+      if [ "#{python}" == "2.7" ] ; then
```
Comment (robzor92) [inline recipes/default.rb]: The code in the if statement is equal to the code in the else statement, can you remove the if-else?

Comment (jimdowling) [reply recipes/default.rb]: Done

**Pull request merged**